### PR TITLE
Fix #29. Add support for protocol-relative URLs

### LINKED ISSUE #29: Support protocol-relative URLs
Currently, it's impossible to create a URL like ```//www.google.com```. Looking into the code, I found out:

```python
# https://github.com/bbangert/routes/blob/master/routes/util.py#L259-L262
if not protocol:
    protocol = config.protocol
if url is not None:
    url = protocol + '://' + host + url
```

That could be changed to accept that if ```protocol == ''```, then we create a protocol-relative URL.

What do you think, @bbangert? I'd be happy to submit a pull request if you agree.

### PR DESCRIPTION
**Pull request by webknjaz**
Fix #29

### CODE REVIEW

**Comment by webknjaz** [comment]
@bbangert ping?

File: routes/util.py [hunk 1]
```diff
@@ -250,21 +254,23 @@ def url_for(*args, **kargs):
             newargs = _screenargs(kargs, config.mapper, environ)
         anchor = newargs.pop('_anchor', None) or anchor
         host = newargs.pop('_host', None) or host
-        protocol = newargs.pop('_protocol', None) or protocol
+        protocol = newargs.pop('_protocol', protocol)
```
Comment (bbangert) [inline routes/util.py]: :+1:

File: routes/util.py [hunk 2]
```diff
@@ -250,21 +254,23 @@ def url_for(*args, **kargs):
...
         url = config.mapper.generate(*route_args, **newargs)
     if anchor is not None:
         url += '#' + _url_quote(anchor, encoding)
-    if host or protocol or qualified:
+    if host or protocol is not None or qualified:
```
Comment (bbangert) [inline routes/util.py]: Can you paren in the (protocol is not None) bit? I've had problems in the past where order of operation binding messes things up in non-obvious ways.

File: routes/util.py [hunk 3]
```diff
@@ -396,12 +405,13 @@ def __call__(self, *args, **kargs):
...
             host = host or newargs.pop('_host', None)
-            protocol = protocol or newargs.pop('_protocol', None)
+            if protocol is None:
+                protocol = newargs.pop('_protocol', None)
             newargs['_environ'] = self.environ
             url = self.mapper.generate(*route_args, **newargs)
         if anchor is not None:
             url += '#' + _url_quote(anchor, encoding)
-        if host or protocol or qualified:
+        if host or protocol is not None or qualified:
```
Comment (bbangert) [inline routes/util.py]: Parens again.

**Comment by bbangert** [comment]
Looks good overall, will merge after the paren change. Thanks!

**Comment by webknjaz** [comment]
Okay, ping me if you will need some conflict resolution my side :)

**Comment by bbangert** [comment]
Just need some parens added in the places indicated.

**Comment by webknjaz** [comment]
Oh, sure!
I didn't mention your comments to diff, since my mobile GitHub client (Gitskarios) doesn't show those for some reason.

**Comment by bbangert** [comment]
Ah, no worries. Thanks for another nicely done patch.

**Pull request merged**